### PR TITLE
Refactor safezone positioning (fix selection outside of canvas)

### LINKED ISSUE #1545: Lasso selection coordinates inaccurate
## Bug Description

Lasso selection coordinates seem to be inaccurate and not following the cursor.

## Expected Behaviour
Lasso selection should be displayed based on the cursor coordinates and movement.

## Steps to Reproduce
1. Try lasso selecting
2. See the outline being created inaccurately

## Screenshots
lasso

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #1787: Element edges/borders incorrect when part of the element is outside of canvas
## Bug Description

If I create an element (shape, images) and drag them down to below the bottom edge of the canvas their selection area is incorrectly placed as starting from the bottom to the canvas up, covering other parts of the canvas where the shape/image is not. 

## Steps to Reproduce

1. Add shape
2. Drag shape partially outside of canvas
3. select element

## Screenshots

Video: 
[Odd selection3.mov.zip](https://github.com/google/web-stories-wp/files/4629531/Odd.selection3.mov.zip)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by wassgha**
## Summary

Minor refactor to the safezone/dangerzone positioning to fix oddities in measurements in react-moveable and lasso selection. Most notably fixes an issue that used to shift frame boxes when selecting an element that partially sits outside of the canvas.

## Relevant Technical Choices

Moved to using `position: absolute` for a few layers to minimize side effects of adding/removing elements to the layers and used explicit pixel heights of the danger zone instead of relying on `flex-box` centering.

## Testing Instructions

1. Add an element to the canvas
2. Drag the element to the edge of the canvas such that part of the element is outside of the page
3. Unselect and reselect the element, this should show the correct box around the element, which used to be wrong.

Fixes #1787 
Fixes #1545

### CODE REVIEW

File: assets/src/edit-story/components/overlay/withOverlay.js [hunk 1]
```diff
@@ -31,8 +31,10 @@ const Overlay = styled.div`
   top: 0;
   /*! @noflip */
   left: 0;
-  width: 0;
-  height: 0;
+  width: 100%;
+  height: 100%;
+  display: block;
+  pointer-events: none;
```
Comment (dvoytenko) [inline assets/src/edit-story/components/overlay/withOverlay.js:37]: Is this ok for all users of `Overlay`?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

**Comment by miina** [comment]
@wassgha It still seems to have the issue in testing, actually, now the issue seems to appear for all the elements -- for the ones that are on the Canvas as well and not partly outside of the edge:
Screenshot 2020-06-05 at 12 41 14

Thoughts?

File: assets/src/edit-story/components/canvas/layout.js [hunk 2]
```diff
@@ -141,7 +138,7 @@ const PageAreaDangerZone = styled.div`
   );
   opacity: 0.05;
   width: 100%;
-  height: calc((var(--fullbleed-height-px) - var(--page-height-px)) / 2);
+  height: var(--danger-zone-height-px);
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/layout.js:141]: Also, this is expressible in `%`, right? I just did something like this in https://github.com/google/web-stories-wp/pull/2265

Comment (dvoytenko) [reply assets/src/edit-story/components/canvas/layout.js:141]: Though, of course, if you need the `--danger-zone-height` anyway, we should just use it.

**Review by dvoytenko — CHANGES_REQUESTED** [review]
Let's hold off this a bit until we figure out the https://github.com/google/web-stories-wp/pull/2285

**Pull request closed without merging**